### ISSUE TITLE
The result in Core is inconsistent with Framework when adding a space for Default namespace

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/project-system#7253]
**Visual Studio Version**: Version 16.11.0 Preview 1.0 [31319.401.d16.11]

**Steps to Reproduce**:

1.	Create a Winforms core application.
2.	Right click project to open properties window.
3.	Find Application page, change Default namespace from WinFormsApp5 to WinForms App5(there is a space), then press enter key.

**Expected Behavior**:
For Framework, an error dialog pops up.
![issue3_fra](https://user-images.githubusercontent.com/45864017/119071016-9fa1da80-ba1b-11eb-80c2-e61f2671458a.png)

**Actual Behavior**:
For Core, it can be changed successfully. 
![Issue3-Core](https://user-images.githubusercontent.com/45864017/119071021-a3cdf800-ba1b-11eb-94dc-724e8fcd5269.gif)

**More info**: 

1. This issue can also repro in .NET Core WPF and console app.
2. This is not a regression issue, it can also repro on 16.7.0 Preview 2.0.

**Comment by drewnoakes** [comment]
We currently perform no UI validation on input strings. We could add some. In #6907 I proposed adding a `RegexPattern` as optional metadata to string properties, which the UI could use to validate inputs. That approach could be a viable solution to this problem.

**Comment by tmeschter** [comment]
I don't think we will fix this for the current property pages but we should consider how we would deal with this for the new property pages.

**Comment by tmeschter** [comment]
One option is to have a property interceptor that strips out the space (or other unsupported characters).

**Comment by drewnoakes** [comment]
cc @adamint as you were looking for examples to apply validation to.
